pkg/adapters/opensearch.go

```diff
@@ -0,0 +1,147 @@
+package adapters
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/accuknox/kubearmor/pkg/models"
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/log"
+	"github.com/kubearmor/KubeArmor/KubeArmor/types"
+	opensearch "github.com/opensearch-project/opensearch-go"
+	"github.com/opensearch-project/opensearch-go/opensearchapi"
+)
+
+type OpenSearchClient struct {
+	client *opensearch.Client
+	index  string
+}
+
+func NewOpenSearchClient(dsc models.DataStoreConfig, allowInsecureTLS bool) (*OpenSearchClient, error) {
+
+	osAddress := dsc.URL
+
+	if osAddress == "" {
+		osAddress = os.Getenv("OS_URL")
+	}
+
+	index := dsc.Index
+	if index == "" {
+		index = os.Getenv("OS_INDEX")
+	}
+
+	cfg := opensearch.Config{
+		Addresses: []string{dsc.URL},
+		Username:  dsc.Username,
+		Password:  dsc.Password,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: dsc.TLSSkipVerify,
+			},
+		},
+	}
+
+	if len(dsc.CACert) >= 0 && dsc.TLSAuthWithCACert {
+		cfg.CACert = dsc.CACert
+	}
+
+	client, err := opensearch.NewClient(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create OpenSearch client: %w", err)
+	}
+
+	osc := &OpenSearchClient{
+		client: client,
+		index:  index,
+	}
+	return osc, nil
+}
+
+func (osc *OpenSearchClient) GetLogs(ctx context.Context, qm models.QueryModel) []types.Log {
+
+	ctxLogger := log.DefaultLogger.FromContext(ctx)
+
+	batchSize := 1000
+	if qm.BatchSize > 0 {
+		batchSize = qm.BatchSize
+	}
+
+	// Prepare search query
+	query := fmt.Sprintf(`{
+  "query": {
+    "wildcard": {
+      "TTY.keyword": "pts*"
+    }
```

Comment:
can you please explain what does this query mean?

Reply:
This queries for logs with  TTY field as we only need to track events with terminal device attached to it to create a process execution graph